pex/cli/commands/lock.py

```diff
@@ -555,25 +563,41 @@ def _export(self):
                 downloaded_artifact.artifact.fingerprint
             )
 
-        with self.output(self.options) as output:
-            pins = fingerprints_by_pin.keys()  # type: Iterable[Pin]
-            if self.options.sort_by == ExportSortBy.PROJECT_NAME:
-                pins = sorted(pins, key=attrgetter("project_name.normalized"))
-            for pin in pins:
-                fingerprints = fingerprints_by_pin[pin]
-                output.write(
-                    "{project_name}=={version} \\\n"
-                    "  {hashes}\n".format(
-                        project_name=pin.project_name,
-                        version=pin.version.raw,
-                        hashes=" \\\n  ".join(
-                            "--hash={algorithm}:{hash}".format(
-                                algorithm=fingerprint.algorithm, hash=fingerprint.hash
-                            )
-                            for fingerprint in fingerprints
-                        ),
+        if self.options.format == ExportFormat.PIP:
+            with self.output(self.options) as output:
+                pins = fingerprints_by_pin.keys()  # type: Iterable[Pin]
+                if self.options.sort_by == ExportSortBy.PROJECT_NAME:
+                    pins = sorted(pins, key=attrgetter("project_name.normalized"))
+                for pin in pins:
+                    fingerprints = fingerprints_by_pin[pin]
+                    output.write(
+                        "{project_name}=={version} \\\n"
+                        "  {hashes}\n".format(
+                            project_name=pin.project_name,
+                            version=pin.version.raw,
+                            hashes=" \\\n  ".join(
+                                "--hash={algorithm}:{hash}".format(
+                                    algorithm=fingerprint.algorithm, hash=fingerprint.hash
+                                )
+                                for fingerprint in fingerprints
+                            ),
+                        )
                     )
-                )
+        else:
+            vertices = set()
```

Comment:
FYI: The current bit of Pex producing graphs is here: https://github.com/pantsbuild/pex/blob/97a2497e0938ece709310a03e7e41b5c26992952/pex/tools/commands/graph.py#L30-L83. That does take markers into account to good effect in the rendered graph, showing which requirements are not active in the present PEX being run by the present interpreter on the present machine. Of course the format is the dot adjacency list format and the side-band information about activation is stored in labels and colors.

If it's still the case that Pants doesn't want to define a format and Pex must my requirements are:
+ Embedding a version for the format: allow an escape hatch for backwards incompatible evolution (but with the intent to try hard to never actually use it)
+ Eventually merge the existing model / logic driving the tool example above with this: one kicks out dot, the other JSON.